src/backend/opencl/Array.cpp

```diff
@@ -51,7 +51,7 @@ template<typename T>
 Array<T>::Array(dim4 dims)
     : info(getActiveDeviceId(), dims, 0, calcStrides(dims),
            (af_dtype)dtype_traits<T>::af_type)
-    , data(bufferAlloc(info.elements() * sizeof(T)), bufferFree)
+    , data(memAlloc<T>(info.elements()).release(), bufferFree)
```

Comment:
This has been bothering me too for a while now - two different ways to allocate buffers in OpenCL.